ui/numbered_menu.py

```diff
@@ -0,0 +1,49 @@
+from ui import Menu
+from ui.base_list_ui import BaseListUIElement
+from ui.utils import clamp
+
+
+class UIWidget(object):
+    pass  # todo : complete
+
+
+class NumberedMenu(UIWidget, Menu):
+    """
+    todo: options:
+    - activate entry after type or select only
+    - delay for keypresses
+    """
+
+    def __init__(self, *args, **kwargs):
+        Menu.__init__(self, *args, **kwargs)
+        self.key_map = {
+            "KEY_1": 1,
+            "KEY_2": 2,
+            "KEY_3": 3,
+            "KEY_4": 4,
+            "KEY_5": 5,
+            "KEY_6": 6,
+            "KEY_7": 7,
+            "KEY_8": 8,
+            "KEY_9": 9,
+            "KEY_0": 0,
+        }
+
+    def activate(self):
+        self.i.set_streaming(self.on_key_pressed)
+        Menu.activate(self)
+
+    def deactivate(self):
```

Comment:
If I'm not mistaken, this is not going to work in case of two nested NumericMenus, as the second Menu's `activate()` is going to overwrite first menu's streaming callback, and once the second menu exits, the first menu numeric callbacks will not be set back. I suggest you just use `set_keymap()`, since that's where the normal keymap is set, too =)